crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -759,4 +823,46 @@ mod tests {
 
         assert_eq!(a, vec![1]);
     }
+
+    #[derive(Component)]
+    struct TestComponent(u32);
+
+    #[test]
+    fn entity_ref_get_by_id() {
```

Comment:
Can you also add a test for a custom component created using `ComponentDescriptor::new_with_layout`?

Reply:
Done (for a custom resource, not a custom component, since component insertion isn't implemented yet)